### PR TITLE
Bug 1213090 - Implement PageWithHeader (attempt to make it)

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e1cfec1438dace1f80d6c8facd5fbf12069f37b

File: tests/python/gaia-ui-tests/gaiatest/apps/messages/regions/new_message.py [hunk 1]
```diff
@@ -151,5 +151,7 @@ def save_as_draft(self):
 
     def go_back(self):
         GaiaHeader(self.marionette, self._header_locator).go_back()
+        # How to handle the fact that sometimes you return a whole new panel,
+        # except by having these lines?
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/messages/regions/new_message.py]: No idea, sounds like a specific thing to only this region.

File: tests/python/gaia-ui-tests/gaiatest/apps/settings/app.py [hunk 2]
```diff
@@ -341,6 +341,7 @@ def _wait_for_toggle_ready(self, by, locator):
     def return_to_prev_menu(self, parent_view, exit_view):
         GaiaHeader(self.marionette, exit_view.find_element(*self._header_locator)).go_back()
 
+        # How to make sure we not in the same panel anymore, except with these lines?
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/settings/app.py]: Isn't that what's being checked then with the following lines?

File: tests/python/gaia-ui-tests/gaiatest/apps/ui_tests/app.py [hunk 3]
```diff
@@ -72,4 +72,5 @@ def tap_keyboard_option(self):
         return KeyboardPage(self.marionette)
 
     def tap_back_button(self):
+        # We're in an app. Inheriting from PageWithHeader doesn't feel right
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/ui_tests/app.py]: So we need AppWithHeader then?

File: tests/python/gaia-ui-tests/gaiatest/apps/base.py [hunk 4]
```diff
@@ -128,7 +128,28 @@ def wait_to_not_be_displayed(self):
...
+
+
+class PageWithHeader(PageRegion):
+
+    def __init__(self, marionette, _header_locator):
+        Base.__init__(self, marionette)
+        self._header_locator = _header_locator
+
+    def go_back(self):
+        self._header.go_back()
```
Comment (zapion) [inline tests/python/gaia-ui-tests/gaiatest/apps/base.py]: This function is supposed to return a page instance and explicit tell caller which page will be displayed then after.

**Comment by JohanLorenzo** [comment]
Mass closing PRs. Gaia is not a supported project anymore.

**Pull request closed without merging**